web/modules/custom/dbcdk_community/src/Form/QuarantineEditForm.php

```diff
@@ -0,0 +1,203 @@
+<?php
+
+/**
+ * @file
+ * Contains \Drupal\dbcdk_community\Form\QuarantineEditForm.
+ */
+
+namespace Drupal\dbcdk_community\Form;
+
+use DBCDK\CommunityServices\Api\QuarantineApi;
+use DBCDK\CommunityServices\ApiException;
+use DBCDK\CommunityServices\Model\Profile;
+use DBCDK\CommunityServices\Model\Quarantine;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+use Drupal\Core\DependencyInjection\ContainerInjectionInterface;
+use Drupal\Core\Form\FormBase;
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\Core\Url;
+
+/**
+ * Edit a Community Quarantine.
+ */
+class QuarantineEditForm extends FormBase implements ContainerInjectionInterface {
+
+  /**
+   * The date format the Community Client expects.
+   */
+  const DATE_FORMAT = 'Y-m-d';
+
+  /**
+   * The Community Service Quarantine object from the context.
+   *
+   * @var \DBCDK\CommunityServices\Model\Quarantine $quarantine
+   */
+  protected $quarantine;
+
+  /**
+   * The Community Service Profile from the Quarantined ID.
+   *
+   * @var \DBCDK\CommunityServices\Model\Profile $profile
+   */
+  protected $profile;
+
+  /**
+   * The DBCDK Community Service Quarantine API.
+   *
+   * @var \DBCDK\CommunityServices\Api\QuarantineApi $quarantineApi
+   */
+  protected $quarantineApi;
+
+  /**
+   * QuarantineEditForm constructor.
+   *
+   * @param \DBCDK\CommunityServices\Model\Quarantine $quarantine
+   *   The Community Quarantine we wish to alter.
+   * @param \DBCDK\CommunityServices\Model\Profile $profile
+   *   The Community Profile we wish to alter a quarantine on.
+   * @param \DBCDK\CommunityServices\Api\QuarantineApi $quarantine_api
+   *   The Community Service Quarantine API.
+   */
+  public function __construct(Quarantine $quarantine, Profile $profile, QuarantineApi $quarantine_api) {
+    $this->quarantine = $quarantine;
+    $this->profile = $profile;
+    $this->quarantineApi = $quarantine_api;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container) {
+    try {
+      /* @var \DBCDK\CommunityServices\Api\QuarantineApi */
+      $quarantine_api = $container->get('dbcdk_community.api.quarantine');
+      $quarantine_filter = [
+        'where' => [
+          'id' => $container->get('request_stack')->getCurrentRequest()->get('quarantine_id'),
+        ],
+      ];
+      $quarantine = $quarantine_api->quarantineFindOne(json_encode($quarantine_filter));
```

Comment:
This has been implemented here: https://github.com/DBCDK/biblo-admin/pull/38/files#diff-f2a84a73f7355c1bbbdd2530354f88e1R72